cherrypy/lib/caching.py

```diff
@@ -411,7 +411,9 @@ def tee(body):
                              body, response.time), len(body))
 
     response = cherrypy.serving.response
-    response.body = tee(response.body)
+    tee_gen = tee(response.body)
+    if tee_gen:
```

Comment:
could you please explain your solution in more detail (and maybe comment/docstring in code)?